File: app/src/main/java/com/ntoprevd/cogno/data/db/dao/SessionDao.java
package com.ntoprevd.cogno.data.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import com.ntoprevd.cogno.data.db.entity.SessionEntity;
import java.util.List;

@Dao
public interface SessionDao {

    @Query("SELECT * FROM sessions ORDER BY pinned DESC, updated_at DESC")
    List<SessionEntity> getAllSessionsOrderByUpdatedAtDesc();

    @Query("SELECT * FROM sessions WHERE id = :id LIMIT 1")
    SessionEntity getSessionById(String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSession(SessionEntity session);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSessions(List<SessionEntity> sessions);

    @Update
    void updateSession(SessionEntity session);

    @Delete
    void deleteSession(SessionEntity session);

    @Query("DELETE FROM sessions WHERE id = :id")
    void deleteSessionById(String id);
}

File: app/src/main/kotlin/com/ntoprevd/cogno/data/db/dao/SessionDao.kt
package com.ntoprevd.cogno.data.db.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.ntoprevd.cogno.data.db.entity.SessionEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface SessionDao {
    @Query("SELECT * FROM sessions ORDER BY pinned DESC, updated_at DESC")
    fun observeAllSessionsOrderByUpdatedAtDesc(): Flow<List<SessionEntity>>

    @Query("SELECT * FROM sessions ORDER BY pinned DESC, updated_at DESC")
    suspend fun getAllSessionsOrderByUpdatedAtDesc(): List<SessionEntity>

    @Query("SELECT * FROM sessions WHERE id = :id LIMIT 1")
    suspend fun getSessionById(id: String): SessionEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertSession(session: SessionEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertSessions(sessions: List<SessionEntity>)

    @Update
    suspend fun updateSession(session: SessionEntity)

    @Delete
    suspend fun deleteSession(session: SessionEntity)

    @Query("DELETE FROM sessions WHERE id = :id")
    suspend fun deleteSessionById(id: String)
}
